src/cloudimagedirectory/transform/transform.py

```diff
@@ -295,27 +295,144 @@ def run(self, data: type[Transformer]) -> list:
+        return results
+
+
+class TransformerAzureV2RHEL(Transformer):
+    """Transform raw rhel Azure data into the schema."""
+
+    def run(self, data):
+        """Transform the raw data."""
+        # NOTE: Verify that the data is raw and provided by azure.
+        entries = [x for x in data if x.is_provided_by("azure") and x.is_raw()]
+
+        results = []
+        for e in entries:
+            entry = copy.deepcopy(e)
+            raw = self.src_conn.get_content(entry)
+
+            for content in raw.content:
+                if content["publisher"] != "RedHat":
+                    continue
+
+                content["hyperVGeneration"] = "unknown"
+
+                image_data = format_azure.image_rhel(content)
+                image_name = image_data["name"].replace(" ", "_").lower()
+                os_name = "rhel"
+                provider = "azure"
+                region = "global"
+                version = image_data["version"]
+                # NOTE: Due to consistency issues between the cloud providers and the fact
+                # that they do not all have unique numbers to identify their images, we decided
+                # to use this solution instead.
+                image_id = hashlib.sha1(image_name.encode()).hexdigest()
+
+                # NOTE: example of expected paths
+                # v2/os/rhel/provider/azure/version/8.6.0/region/southcentralus/image/71d0a7aaa1f0dc06840e46f6ce316a7acfb022d4
+                # v2/os/rhel/provider/azure/version/8.2.0/region/southcentralus/image/14e4eab326cc5a2ef13cb5c0f36bc9bfa41025d9
+                path = f"v2/os/{os_name}/provider/{provider}/version/{version}/region/{region}/image/{image_id}"
+                data_entry = connection.DataEntry(path, image_data)
+
+                results.append(data_entry)
+        return results
+
+
+class TransformerGoogleV2RHEL(Transformer):
+    """Transform raw rhel Google data into the schema."""
+
+    def run(self, data):
+        """Transform the raw data."""
+        # NOTE: Verify that the data is raw and provided by google.
+        entries = [x for x in data if x.is_provided_by("google") and x.is_raw()]
+
+        results = []
+        for e in entries:
+            entry = copy.deepcopy(e)
+            raw = self.src_conn.get_content(entry)
+
+            for content in raw.content:
+                content["creation_timestamp"] = content["creationTimestamp"]
+                if "rhel" in content["name"]:
+
+                    image_data = format_google.image_rhel(content)
+                    image_name = image_data["name"].replace(" ", "_").lower()
+                    region = "global"
+                    os_name = "rhel"
+                    provider = "google"
+                    version = image_data["version"]
+                    # NOTE: Due to consistency issues between the cloud providers and the fact
+                    # that they do not all have unique numbers to identify their images, we decided
```

Comment:
Are those names that you're using for hashing unique? 
Just asking cause the created hash will be deterministic. 
Not that we're running into the same id overlap with those hashes.

Reply:
Not in general unique, but in their specific path, they are.

Before we had the identification based on the image name. Now it's based on the full path. 
For example if we have two equal image names `XXXXXXXXXXXXXXXXXXXXXX` and those are different images, it should have no conflict because they are stored in different paths :) 

like this:
```bash
/v2/os/rhel/provider/azure/version/7-8/region/eu-west-1/image/XXXXXXXXXXXXXXXXXXXXXX
/v2/os/fedora/provider/aws/version/9/region/us-east-22/image/XXXXXXXXXXXXXXXXXXXXXX
````

Reply:
Okay cool! I guess we're not running into any edge cases in which we need to lookup images by their imageId value alone, right?

Reply:
Not so far. In that case we would need another endpoint to lookup images by ID. For example `/v2/image-ids/list`, right?
That would return a list of image IDs and their full paths.

Reply:
Yeah, as long as we avoid that we should be fine.